all-the-icons.el

```diff
@@ -868,10 +867,30 @@ inserting functions."
     (when arg-overrides (setq args (append `(,(car args)) arg-overrides (cdr args))))
     (apply (car icon) args)))
 
-(memoize 'all-the-icons-icon-for-dir)
-(memoize 'all-the-icons-icon-for-file)
-(memoize 'all-the-icons-icon-for-mode)
-(memoize 'all-the-icons-icon-for-url)
+(defconst all-the-icons--cache-limit 2048
+  "Max cache size for functions made cached by `all-the-icons-cache'.")
+
+(defun all-the-icons-cache (func)
+  "Set a cache for FUNC. Does not work on interactive functions."
+  (unless (get func 'all-the-icons--cached)
+    (let ((cache (make-hash-table :test #'equal
+                                  :size all-the-icons--cache-limit))
+          (orig-fn (symbol-function func)))
+      (fset func
+            (lambda (&rest args)
+              (or (gethash args cache)
+                  (progn
+                    (when (> (hash-table-count cache)
+                             all-the-icons--cache-limit)
+                      (clrhash cache))
```

Comment:
This is undesirable as it will present a massive performance hit for packages like `all-the-icons-dired`, can you change this implementation to use some kind of LRU policy?

https://en.wikipedia.org/wiki/Cache_replacement_policies

Reply:
I'll try, but it needs experimentation. I'm not sure a complicated LRU cache would be faster than a plain hash map.